.circleci/scripts/pre_commit_setup_cfg_python_3_11_all_extra.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+"""
+Pre-commit hook to sync a "test_python_3_11" extra in setup.cfg.
+It will contain all the dependencies apart from tests and mypy.
+"""
+import configparser
+from pathlib import Path
+
+repo_dir = Path(__file__).parent.parent.parent
+
+config = configparser.ConfigParser(strict=False)
+config.read(repo_dir / "setup.cfg")
+
+all_extra = set()
+extra_to_exclude = {"tests", "mypy", "docs", "all", "test_python_3_11", "apache.hive"}
+for key, extra_value in config["options.extras_require"].items():
+    if key in extra_to_exclude:
+        continue
+    reqs = extra_value.split()
+    all_extra |= set(reqs)
+
+expected_test_python_3_11_extra = all_extra
```

Comment:
I think we might not need `all_extra`. Also, I notice the comment is not correctly parsed

```suggestion
expected_test_python_3_11_extra = {
    req
    for key, extra_value in config["options.extras_require"].items()
    for req in extra_value.split()
    if key not in extra_to_exclude
}

```